### DISCUSSION TITLE
Make tracker statistics available via prometheus

### DISCUSSION THREAD

**Discussion by ldpr** [torrust/torrust-tracker#135] [Feature requests]
Currently the tracker stats are output in json, it'd be pretty neat if we could have them available via prometheus so we could hook them into things like grafana etc.

https://docs.rs/prometheus_exporter/latest/prometheus_exporter/

Maybe we could have a config option for json _or_ prometheus?

Obviously this isn't important or anything, would be a "nice to have". Thanks for everything you do guys and girls! :)

**Comment by josecelano** [comment]
I think persisting statistics will also be a nice feature. Now statistics are reset when you restart the service.

**Reply by josecelano** [reply]
I found this docker compose configuration:

https://github.com/bitmagnet-io/bitmagnet/blob/32dc0bf6ec8db0a4aea0be1c59d7c8a17b13f268/docker-compose.yml

that might be useful to setup prometheus and grafana.

**Reply by josecelano** [reply]
I like what https://www.gbitt.info/ does:

![image](https://github.com/torrust/torrust-tracker/assets/58816/ab4b3e4c-a2dd-4521-85fe-c9521e76c817)

**Comment by josecelano** [comment]
Hi, @ldpr; this has been implemented this week. See https://github.com/torrust/torrust-tracker/issues/1126. I have deployed it to the live demo and created this sample dashboard

![image](https://github.com/user-attachments/assets/d57458c1-1a69-4a69-b021-1aae7677134f)

I'm planning to write a blog post about it: https://github.com/torrust/torrust-website/issues/131

And also to add some more metrics: https://github.com/torrust/torrust-tracker/issues/1128

cc @da2ce7
